### PR TITLE
Adapt Haddock to Logger and Parser changes

### PR DESCRIPTION
**Pull request by hsyl20**
Companion PR to https://gitlab.haskell.org/ghc/ghc/-/merge_requests/5845

### CODE REVIEW

**Review by alexbiehl — APPROVED** [review]
Thanks sir! 🚢

File: haddock-api/src/Haddock.hs
Comment (hsyl20) [inline haddock-api/src/Haddock.hs:303]: Just noticed there is an issue here. Please don't merge!

Comment (hsyl20) [reply haddock-api/src/Haddock.hs]: Fixed!

**Comment by Kleidukos** [comment]
I just got the green light from @hsyl20! Let's go! :rocket:

**Pull request merged**
